docs/metastore_upgrade_guide.md

```diff
@@ -0,0 +1,213 @@
+
+The CLI command will create a mapping file in the workspace UCX folder.
+The mapping file is in CSV format and can be edited using any text editor or Excel.
+
+The format of the mapping file is as follows:
+
+| **columns:** | **workspace_name**      | **catalog_name** | **src_schema** | **dst_schema** | **src_table** | **dst_table** |
+|--------------|---------------------|--------------|----------------|----------------|---------------|---------------|
+| values:      | data_engineering_ws | de_catalog   | database1      | database1      | table1        | table1        |
+
+The
+
+### Step 2: Create the necessary cloud principals for the upgrade
+This step has multiple sub steps and should be performed by a cloud account administrator.
+We rely on CLI access (AWS) or API Access (Azure) to create the necessary cloud principals.
+In the future we will add support for creating the necessary cloud principals using Terraform.
+
+We refer to principals that has to be created/modified during the process.
+Principal are:
+Roles/Instance Profiles in AWS
+Service Principals/Managed Identities in Azure
+
+To understand this step it is important to understand how Databricks accesses cloud locations.
+You can read about it here:
+[AWS - Create External Locations](https://docs.databricks.com/en/connect/unity-catalog/external-locations.html)
+[Azure - Create External Locations](https://learn.microsoft.com/en-us/azure/databricks/connect/unity-catalog/external-locations)
+
+#### Step 2.1: Map the cloud principals to the cloud "prefixes"
+In this step we are going to map all the cloud principals to the paths they have access to.
+
+
+#### Step 2.2: Create/Modify Cloud Principals and Credentials
+In this step we will create the necessary cloud principals for the UC credentials.
+The manual process is documented in the following links:
+[AWS-Storage Credentials](https://docs.databricks.com/en/connect/unity-catalog/storage-credentials.html)
+[Azure-Storage Credentials](https://learn.microsoft.com/en-us/azure/databricks/sql/language-manual/sql-ref-storage-credentials)
+
+For AWS we have the to create fresh new AWS roles and set them up for UC access, using the following command:
+```text
+databricks labs ucx create-missing-principles --aws-profile <aws_profile> --single-role <single_role>
+```
+This command identifies all the S3 locations that are missing a UC compatible role and creates them. 
+It takes single-role optional parameter. 
+If set to True, it will create a single role for all the S3 locations.
+Otherwise, it will create a role for each S3 location.
+
+
+For both AWS and Azure we can use the following CLI command to upgrade the necessary cloud principals:
+```text
+databricks labs ucx migrate-credentials
+```
+
+For both AWS and Azure this command will create the necessary UC storage credentials in the UC metastore.
+
+Azure: this command migrates Azure Service Principals, which have Storage Blob Data Contributor,
```

Comment:
Also looks for SP with custom roles and it prompts to ask for creating access connectors. Note that there is a risk here for the two documentations to become out-of-sync